File: LLD/LLD-2/src/TicTakToeGame/src/main/java/org/example/models/Board.java
package org.example.models;

import java.util.ArrayList;
import java.util.List;

public class Board {
    private int size;

    private List<List<Cell>> board;

    Board(int dimension)
    {
        //Initialize a board of size n

        this.size = dimension;
        board = new ArrayList<>(); // addind rows
        for(int i=0;i<size;i++)
        {
            board.add(new ArrayList<>());// adding columns
            for(int j=0;j<size;j++)
            {
                board.get(i).add(new Cell(i,j)); //adding cells for column
            }
        }
    }

    public void printBoard()
    {
        for(int i=0;i<size;i++)
        {
            for(int j=0;j<size;j++)
            {
                Cell cell = board.get(i).get(j);
                if(cell.getCellState().equals(CellState.EMPTY))
                    System.out.print("|  |");
                else
                    System.out.print("| " +cell.getPlayer().getSymbol().getSymbol() + " |");
            }
            System.out.println();
        }
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public List<List<Cell>> getBoard() {
        return board;
    }

    public void setBoard(List<List<Cell>> board) {
        this.board = board;
    }
}

File: LLD3/TicTacToe/TicTacToe/Models/Board.cs
using System;
using System.Collections.Generic;

namespace TicTacToe.Models
{
    public class Board
    {
        private Cell[,] _board;
        private int _dimension;

        public Board(int dimension)
        {
            _dimension = dimension;
            _board = new Cell[_dimension,_dimension];
            for(int i=0;i< _dimension;i++)
            {
                for(int j=0;j< _dimension;j++)
                {
                    _board[i,j] = new Cell(i,j);
                }
            }

        }

        public int Dimension { get => _dimension; set => _dimension = value; }
        public Cell[,] Boards { get => _board; }


        /*
         *  --------------
            |   |   |   |
            --------------
            |   |   |   |
            --------------
            |   |   |   |
            --------------
        */
        public void DisplayBoard()
        {
            Console.Write("------------------\n");
            

            for(int i=0;i<_dimension;i++)
            {
                for(int j=0;j<_dimension;j++)
                {
                    Cell cell = _board[i,j];
                    if(cell.State.Equals(CellState.Empty))
                    {
                        Console.Write("|  |");
                    }
                    else
                    {
                        Console.Write("|" + cell.Player.Symbol + "|");
                    }
                }
                Console.Write("\n");
                Console.Write("------------------\n");
            }
            Console.Write("\n");
        }
    }
}